include/rmm/mr/device/cuda_async_memory_resource.hpp

```diff
@@ -60,7 +60,21 @@ class cuda_async_memory_resource final : public device_memory_resource {
     posix_file_descriptor = 0x1,  ///< Allows a file descriptor to be used for exporting. Permitted
                                   ///< only on POSIX systems.
     win32     = 0x2,              ///< Allows a Win32 NT handle to be used for exporting. (HANDLE)
-    win32_kmt = 0x4  ///< Allows a Win32 KMT handle to be used for exporting. (D3DKMT_HANDLE)
+    win32_kmt = 0x4,  ///< Allows a Win32 KMT handle to be used for exporting. (D3DKMT_HANDLE)
+    fabric    = 0x8   ///< Allows a fabric handle to be used for exporting. (cudaMemFabricHandle_t)
+  };
+
+  /**
+   * @brief Flags for specifying the memory pool accessibility from other devices.
+   *
+   * @note The default, `prot_none`, marks the pool's memory as private to the device in
+   * which it was created. `prod_read_write` should only be used if memory sharing among
+   * devices is required. Note that there is a `cudaMemAccessFlagsProtRead` documented,
+   * but memory pools don't support read-only access, so it has been omitted.
+   */
+  enum class access_flags {
+    prot_none       = 0,  ///< Default, make pool not accessible. (cudaMemAccessFlagsProtNone)
+    prot_read_write = 3   ///< Make pool read-write accessible. (cudaMemAccessFlagsProtReadWrite)
   };
```

Comment:
Note that `read` is not supported for pools. Do we want to expose this flag if it can cause a CUDA error (not supported)?

Reply:
Ah, OK, let's not expose read then.

Reply:
Ok, never mind. I can't change `allocation_handle_type` in this case, because I can't compile it with older versions of CUDA that rapids supports (11.2+).

Reply:
I'd like to punt on the enum change. Given the logic above, we have two enums that both could follow the underlying, but in the case of fabric, we don't even have that definition so we can't compile. I'd rather follow the pattern that was set there before.

Reply:
OK, makes sense. Please add a comment that these values must be kept in line with the matching values in cudart.

Reply:
Done!